src/platform/packages/shared/kbn-esql-validation-autocomplete/src/shared/helpers.ts

```diff
@@ -755,7 +755,8 @@ export function correctQuerySyntax(_query: string, context: EditorContext) {
     // monaco.editor.CompletionTriggerKind['Invoke'] === 0
     (context.triggerKind === 0 && unclosedRoundBracketCount === 0) ||
     (context.triggerCharacter === ' ' && isMathFunction(query, query.length)) ||
-    isComma(query.trimEnd()[query.trimEnd().length - 1])
+    isComma(query.trimEnd()[query.trimEnd().length - 1]) ||
+    /\|\s*lookup\s+$/i.test(query)
```

Comment:
Not pretty but unfortunately necessary at the moment :(

If we don't add the editor marker to `... | LOOKUP `, the parser blows up instead of nicely informing us that we are missing `JOIN`.

The editor marker is a yucky thing.